core/statix.multilang/src/main/java/mb/statix/multilang/ALanguageMetadata.java

```diff
@@ -0,0 +1,26 @@
+package mb.statix.multilang;
+
+import mb.pie.api.Function;
+import mb.pie.api.TaskDef;
+import mb.resource.ResourceKey;
+import mb.resource.ResourceRegistry;
+import mb.resource.hierarchical.ResourcePath;
+import mb.statix.multilang.spec.SpecBuilder;
+import org.immutables.value.Value;
+import org.spoofax.interpreter.terms.IStrategoTerm;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Value.Immutable
+public abstract class ALanguageMetadata {
```

Comment:
Why not have this be an interface, just call it `LanguageMetadata`, and use that everywhere (except the place where you instantiate an object of this interface?) See the usage of immutables in the `spoofax.compiler` project, in the `mb.spoofax.compiler.spoofaxcore` package.